htdocs/core/lib/functions.lib.php

```diff
@@ -95,6 +95,26 @@ function getEntity($element, $shared = 1, $currentobject = null)
 	}
 }
 
+/**
+ * 	Set entity id to use when to create an object
+ *
+ * 	@param	object	$currentobject	Current object
+ * 	@return	mixed					Entity id to use ( eg. entity = '.setEntity($object) )
+ */
+function setEntity($currentobject)
+{
+	global $conf, $mc;
+
+	if (is_object($mc))
+	{
+		return $mc->setEntity($currentobject);
+	}
+	else
+	{
+		return $conf->entity;
```

Comment:
Here we should have
return (is_object($currentobject) && $currentobject->id > 0) ? $currentobject->entity : $conf->entity;

When currentobject is not an object or no id yet, we are in create mode, so we use $conf->entity
When currentobject is an object and there is an id already, we are in update mode, so we use $conf->entity

So when you replaced $this->entity with setEntity($this), you have same code than before.

Reply:
@eldy yes you right, i send a fix